crates/ruff/src/commands/graph.rs

```diff
@@ -0,0 +1,159 @@
+use crate::args::{ConfigArguments, GraphArgs};
+use crate::resolve::resolve;
+use crate::{resolve_default_files, ExitStatus};
+use anyhow::Result;
+use log::{debug, warn};
+use ruff_graph::{Direction, ImportMap, ModuleDb, ModuleImports};
+use ruff_linter::warn_user_once;
+use ruff_python_ast::{PySourceType, SourceType};
+use ruff_workspace::resolver::{python_files_in_path, ResolvedFile};
+use rustc_hash::FxHashMap;
+use std::collections::BTreeMap;
+use std::path::Path;
+use std::sync::Arc;
+
+/// Generate an import map.
+pub(crate) fn graph(args: GraphArgs, config_arguments: &ConfigArguments) -> Result<ExitStatus> {
+    // Construct the "default" settings. These are used when no `pyproject.toml`
+    // files are present, or files are injected from outside the hierarchy.
+    let pyproject_config = resolve(config_arguments, None)?;
+
+    // Find all Python files.
+    let files = resolve_default_files(args.files, false);
+    let (paths, resolver) = python_files_in_path(&files, &pyproject_config, config_arguments)?;
```

Comment:
Too bad that red knot doesn't support exclude/include configurations yet because you would then get all of this for free (lazy file discovery)